utils/utf8.cc

```diff
@@ -87,8 +95,9 @@ static inline std::optional<size_t> validate_naive(const uint8_t *data, size_t l
                      (byte1 == 0xED && byte2 <= 0x9F) ||
                      // EE..EF, 80..BF, 80..BF
                      (byte1 >= 0xEE && byte1 <= 0xEF))) {
-                bytes = 3;
-            } else if (len >= 4) {
+                return codepoint_status{.bytes_validated = codepoint_len};
+            }
+            {
                 // Is byte4 between 0x80 ~ 0xBF
```

Comment:
Again, compares only against 0xBF. 0x80 is never checked.

Reply:
Again, the codeeis tricky but correct.

Reply:
True.